src/main/java/uk/gov/hmcts/reform/prl/services/PaymentRequestService.java

```diff
@@ -86,4 +102,71 @@ public PaymentStatusResponse fetchPaymentStatus(String authorization,
                                 paymentReference
             );
     }
+
+
+    public PaymentResponse createPayment(@RequestHeader(HttpHeaders.AUTHORIZATION) String authorization,
+                                         @RequestHeader(SERVICE_AUTH) String serviceAuthorization,
+                                         @RequestBody CreatePaymentRequest createPaymentRequest)
+        throws Exception {
+
+        //Get case using Caseid
+        String caseId = createPaymentRequest.getCaseId();
+        uk.gov.hmcts.reform.ccd.client.model.CaseDetails caseDetails = coreCaseDataApi.getCase(
+            authorization,
+            serviceAuthorization,
+            caseId
+        );
+        log.info("Case Data retrieved for id : " + caseDetails.getId().toString());
+        CaseData tempCaseData = CaseUtils.getCaseData(caseDetails, objectMapper);
+        String paymentServiceReferenceNumber = tempCaseData.getPaymentServiceRequestReferenceNumber();
+        String paymentReferenceNumber = tempCaseData.getPaymentReferenceNumber();
+        log.info("paymentServiceReferenceNumber : {}, paymentReferenceNumber :{} for the case id: {} ",
+            paymentServiceReferenceNumber,paymentReferenceNumber,caseId);
+        //Check if paymentServiceReferenceNumber and PaymentReference exist and if yes get the status of payment.
+        //If its success then payment is success else create payment
+        if (paymentServiceReferenceNumber != null && paymentReferenceNumber != null) {
```

Comment:
I think we need to also have an else  if payment Service Ref is present and paymentreferencenumber is not present

Reply:
Fixed